From 5f321b36381d87caf54746254050c33aa28fcfa1 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Tue, 18 Feb 2020 13:54:56 +0200
Subject: [PATCH] [SW-8229] goya2: fix high bandwidth pcie range-register

PCIe doens't have access to PSOC_SECURITY on high bandwidth bus. Fix
ranges to allow only flash related registers, depending on efuse
configuration.

Change-Id: I79556dd1a83241c000ce998e9ae0932cd50f33dd
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 692258d..55e64d8 100644
--- a/common/security.c
+++ b/common/security.c
@@ -232,18 +232,10 @@
 		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
 	}
 
-	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
 	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
+		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
 	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
+		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
 }
 
 static void secuirty_dcon_if1_rr_hbw_config(uint8_t spi_controller_en)
@@ -273,18 +265,10 @@
 		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
 	}
 
-	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
 	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
+		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
 	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
+		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
 }
 
 static void secuirty_i2c_s_rr_hbw_config(uint8_t spi_controller_en)
